transmogrifier/sources/ead.py

```diff
@@ -95,22 +135,17 @@ def get_main_titles(cls, xml: Tag) -> list[str]:
         Args:
             xml: A BeautifulSoup Tag representing a single EAD XML record.
         """
-        main_titles = []
-        if collection_description := xml.metadata.find("archdesc", level="collection"):
-            collection_description_did = collection_description.find("did")
-            if collection_description_did:
-                main_titles = [
-                    unittitle_value
-                    for unittitle_elem in collection_description_did.find_all(
-                        "unittitle", recursive=False
-                    )
-                    if (
-                        unittitle_value := ", ".join(
-                            string for string in unittitle_elem.stripped_strings
-                        )
-                    )
-                ]
-        return main_titles
+        try:
+            unit_titles = xml.metadata.find(
+                "archdesc", level="collection"
+            ).did.find_all("unittitle")
+        except AttributeError:
+            return []
+        return [
+            title.strip()
+            for unit_title in unit_titles
+            if (title := cls.create_string_from_mixed_value(unit_title, "", ["num"]))
```

Comment:
Using `", "` as a separator didn't seem appropriate given the likely non-skipped elements we'll find in `unittitle`s but a one-size-fits-all approach may lead to occasional weirdness